src/overloads/arrays.jl

```diff
@@ -114,6 +114,56 @@ for (Tx, TA, Ty) in Iterators.filter(
     @eval LinearAlgebra.dot(x::$Tx, A::$TA, y::$Ty) = LinearAlgebra.dot(x, A * y)
 end
 
+## Multiplication
+function Base.:*(A::Matrix{T}, B::Matrix) where {T<:AbstractTracer}
```

Comment:
We probably only want this on `GradientTracer`.

Reply:
Why? I think this is also correct for Hessian tracing.

Reply:
Exactly. 👍